### ISSUE TITLE
Game crashes in editor when selecting "autodetect packages" in the save menu

### ISSUE THREAD

**Issue by williethewolf** [Regalis11/Barotrauma#11893]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

CTD on pressing autodetect packages in the subeditor on a sub created on a previous version of the game with mods.

Game version 1.0.9.0 (ReleaseWindows, branch release/1.0-hotfix-2, revision 64dcc9f061)
Graphics mode: 3840x2160 (BorderlessWindowed)

System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 1080 Ti
    Display mode: {Width:3840 Height:2160 Format:Color AspectRatio:1.7777778}
    GPU status: Normal

### Reproduction steps

1.open sub editor
2.load sub from older version with some mods removed and updated
3. Go to save and select auto detect packages.
4. Crash.

### Bug prevalence

Happens every time I play

### Version

v1.0.9.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Barotrauma Client crash report (generated on 04/22/2023 22:42:31)

Barotrauma seems to have crashed. Sorry for the inconvenience! 

8BE661744288B03127E103F2FAFCC948

Game version 1.0.9.0 (ReleaseWindows, branch release/1.0-hotfix-2, revision 64dcc9f061)
Graphics mode: 3840x2160 (BorderlessWindowed)
VSync ON
Language: English
Selected content packages: Vanilla, C4M4-20N, C4MA-20N, Deathstar, E1-Su8, E1-Su8-v1, E1-Su8_BK, El-SuB, El-SuB_1, Handcuff case full, Subphear, Toolbox Stocked, Sweet Pea, Indra, REALLYBIGMAP, Improved Husks, Let Me Choose !, Salty Dogs, Shipwrecks Extended, U-1211, Seated Operators, Toolbox Expanded, [BP] Smart Systems Pack, NUCLEAR U-31 MK2, Apo-4A "Saber", BaroTraumatic, Lua For Barotrauma, Barotraumatic Xanmonsters1 Fixes, Barotraumatic Xanmonsters2 Fixes, Barotraumatic Extended Monsters Fixes, Barotraumatic Survivator Fixes, Barotraumatic Other Creature fixes, Barotraumatic Monster Pack Test fixes, [VH] The Iroh, Ultra Optimized Perfect Airlock, Super Plague Rat, Shreksophone, The Peregrine, Barotraumatic Vanilla Creature Overhauls, Undersea Horrors, CrouchHarder, Transparent Diving Helmets, Apo-5 "Chainsaw", Barotraumatic New Creatures, R-278 «Dugong»
Level seed: no level loaded
Loaded submarine: E1-Su8 Open Access (AEED753C9D12B44AA39E84747686EC92)
Selected screen: Barotrauma.SubEditorScreen
SteamManager initialized

System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 1080 Ti
    Display mode: {Width:3840 Height:2160 Format:Color AspectRatio:1.7777778}
    GPU status: Normal

Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Barotrauma.ContentPackage <CreateSaveScreen>b__128_60(Barotrauma.MapEntity)
Stack trace: 
   at Barotrauma.SubEditorScreen.<>c.<CreateSaveScreen>b__128_60(MapEntity e) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Screens\SubEditorScreen.cs:line 2850
   at System.Linq.Enumerable.SelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.DistinctIterator`1.MoveNext()
   at System.Linq.Enumerable.OfTypeIterator[TResult](IEnumerable source)+MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToHashSet[TSource](IEnumerable`1 source, IEqualityComparer`1 comparer)
   at Barotrauma.SubEditorScreen.<>c__DisplayClass128_0.<CreateSaveScreen>b__59(GUIButton button, Object o) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Screens\SubEditorScreen.cs:line 2850
   at Barotrauma.GUIButton.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUIButton.cs:line 263
   at Barotrauma.GUI.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUI.cs:line 1247
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 906
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 93
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58

Last debug messages:
[04/22/2023 22:41:28] WARNING: Character "Human" has a negative crush depth. Previously the crush depths were defined as display units (e.g. -30000 would correspond to 300 meters below the level), but now they're in meters (e.g. 3000 would correspond to a depth of 3000 meters displayed on the nav terminal). Changing the crush depth from -30000 to 1300.
[04/22/2023 22:41:27] WARNING: Could not compress a texture because the dimensions aren't a multiple of 4 (path: C:/Users/thewo/AppData/Local/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed/2181819352/Item/Cage.png, size: 500x250)
[04/22/2023 22:41:14] WARNING: Character "Human" has a negative crush depth. Previously the crush depths were defined as display units (e.g. -30000 would correspond to 300 meters below the level), but now they're in meters (e.g. 3000 would correspond to a depth of 3000 meters displayed on the nav terminal). Changing the crush depth from -30000 to 1300.
[04/22/2023 22:37:56] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "rubber"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:56] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "titaniumaluminiumalloy"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:55] WARNING: Could not compress a texture because the dimensions aren't a multiple of 4 (path: C:/Users/thewo/AppData/Local/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed/2726528805/Items/Medical/contaminationv1.png, size: 129x128)
[04/22/2023 22:37:55] WARNING: Unrecognized element in affliction "cytokine" (Affliction)
[04/22/2023 22:37:55] WARNING: Unrecognized element in affliction "cytokine" (Affliction)
[04/22/2023 22:37:55] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "rubber"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:55] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "titaniumaluminiumalloy"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:55] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "rubber"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:55] WARNING: Potential error in item variant "slipsuit": the item inherits the fabrication requirement of x1 "titaniumaluminiumalloy"  from the base item "combatdivingsuit". If this is not intentional, you can use empty <RequiredItem /> elements in the item variant to remove any excess inherited fabrication requirements.
[04/22/2023 22:37:54] Overriding an affliction or a buff with the identifier 'watchersgaze' using the file 'C:/Users/thewo/AppData/Local/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed/2666048559/Afflictions.xml'
[04/22/2023 22:37:38] Overriding an affliction or a buff with the identifier 'huskinfection' using the file 'C:/Users/thewo/AppData/Local/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed/2085783214/Afflictions.xml'
```

**Comment by Regalis11** [comment]
Thank you for the report!

I unfortunately couldn't find a way to reproduce this, even though I tried various submarines that require mods, and enabling/disabling those mods. I added a null check on the line mentioned in the crash report though, which should hopefully fix the issue (commit in our private repo https://github.com/Regalis11/Barotrauma-development/commit/ea5fe769ce33ef5f3ce05a3cb43aadfb71d82d44).
